docs/reference/challenges.rst

```diff
@@ -0,0 +1,119 @@
+==========
+Challenges
+==========
+
+Challenge resources are used by the ACME issuer to manage the lifecycle of an
+ACME 'challenge' that must be completed in order to complete an 'authorization'
+for a single DNS name/identifier.
+
+When an **Order** resource is created, the order controller will create
+Challenge resources for each DNS name that is being authorized with the ACME
+server.
+
+As an end-user, you will never need to manually create a Challenge resource.
```

Comment:
not for this PR, but could we disallow creating challenges via RBAC?

Maybe a line or 2 here, telling people that challenge resources exists only to track state and provide debugging